packages/app/src/explorer/utils.ts

```diff
@@ -26,3 +26,22 @@ export function needsNxBadge(
 
   return false;
 }
+
+export function getActiveButtonIndex(
+  element: Element | null
+): [number, HTMLButtonElement[]] | undefined {
+  if (!element) {
+    return undefined;
+  }
+
+  const activeButton = document.activeElement;
```

Comment:
What happens when the viewer renders for the first time? Is `activeElement` defined? If not, then perhaps it should default to the first button (i.e. the button of the root entity)?

Reply:
`activeElement` is indeed `undefined` at the first render. But I don't think it causes a problem: the callback can only be called once the focus is on a the Explorer, that is to say, one element is focused.

In this case, `activeButton` will always be defined. Unless I missed something ? :thinking:

Reply:
Ah right, you put the `keydown` listener on the explorer. Well, in that case, I would suggest moving it up to the `document` so that people can interact with the keyboard to navigate the explorer when the app first renders, and more generally any time they want.

We just have to be careful not to interfere with the keyboard interactions of other UI elements, like the slider, and basically anything else that can be interacted with (drop-down menus, button groups maybe (to be checked), etc.)

Reply:
I think a condition like `!activeElement || explorerRoot.contains(activeElement)` in the key handlers for the explorer should take care of avoiding those interference in a generic way.